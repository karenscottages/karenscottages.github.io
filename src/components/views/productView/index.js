export * from './productView.jsx';
export * from './groupView.jsx';