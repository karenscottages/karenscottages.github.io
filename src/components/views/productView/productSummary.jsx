import React from 'react';
import { Card } from '/components/atoms/cards';
import { LabelValuePair } from '/components/atoms/labelValuePair';
import { ObjectName } from '/agility/structureNode/components';
import { StructureNodeContext } from '/agility/structureNode';

/* ID: 10096
Object Type: Product (2)
Name Path: AreUSerious.co Main Website/Musical Instruments/Rock Band/Guitars/Electric Guitars/Gibson/Gibson SG/123-005
Id Path: /30191/10075/10841/10076/17955/10079/10082/10096
Structure ID: 226
Last Modified: 2019-05-07 08:44:31.0930
Last Modified By: Mark Fee (Steve G)
// TODO Modified and Modified By
*/


export const ProductSummary = () => {
    const { structureNode } = React.useContext(StructureNodeContext);

    return <Card>
        <Card.Header title="Summary" />
        <Card.Content>
            {structureNode && <>

                <ObjectName />

                <LabelValuePair className="is-2 has-text-weight-semibold" label="ID">
                    {structureNode.id}
                </LabelValuePair>

                <LabelValuePair className="is-2 has-text-weight-semibold" label="Object Type">
                    {structureNode.objectType?.name} ({structureNode.objectTypeId})
                </LabelValuePair>

                <LabelValuePair className="is-2 has-text-weight-semibold" label="Name Path">
                    {structureNode.namePath}
                </LabelValuePair>

                <LabelValuePair className="is-2 has-text-weight-semibold" label="Id Path">
                    {structureNode.idPathStr}
                </LabelValuePair>

                <LabelValuePair className="is-2 has-text-weight-semibold" label="Structure Id">
                    {structureNode.structId}
                </LabelValuePair>

            </>}
        </Card.Content>
    </Card>;
};
