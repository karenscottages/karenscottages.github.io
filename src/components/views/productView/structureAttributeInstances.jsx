import React from 'react';
import { AcmsValue } from '/agility/acms/components';
import { LabelValuePair } from '/components/atoms/labelValuePair';

const StructureAttributeInstance = ({attributeInstance}) => 
    <LabelValuePair className="is-narrow has-text-weight-semibold" label={attributeInstance.name || "NONAME"}>
        <AcmsValue acms={attributeInstance.value} />
    </LabelValuePair>;

export const StructureAttributeInstances = ({ structureNode }) => structureNode.attributes.
    filter((attributeInstance) => (attributeInstance.dataTypeId <= 3 && attributeInstance.value)).
    map((attributeInstance) => <StructureAttributeInstance
        key={attributeInstance.name}
        attributeInstance={attributeInstance} />);
