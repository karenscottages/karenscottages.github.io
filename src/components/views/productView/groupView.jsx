import React from 'react';
import { ProductSummary } from './productSummary.jsx';
import { ObjectName, ObjectPreview } from '/agility/structureNode/components';
import { StructureNodeContext } from '/agility/structureNode';
import { Card } from '/components/atoms/cards';
import { useHistory  } from 'react-router-dom';

const GroupView = () => {
    const { structureNode, setIdPath } = React.useContext(StructureNodeContext);
    const [children, setChildren] = React.useState([]);
    const history = useHistory();

    // Effects
    React.useEffect(() => { // Requests the children when a node is expanded
        if (structureNode) {
            structureNode.fetchChildren().then(() => setChildren(structureNode.children));
        } else {
            setChildren([]);
        }
    }, [structureNode]);

    const onSelectChild = (structureNode) => {
        setIdPath(structureNode.idPathStr);
        history.push("/productview"); // TODO this should be dynamic like the structureBrowser
    };

    return <div className="container is-fluid">
        <div className="section columns is-multiline">

            <div className="column">
                <ProductSummary />
            </div>

            {children && <div className="column">
                <Card>
                    <Card.Header>Children</Card.Header>
                    <Card.Content>{
                        children.map((childStructureNode) => 
                            <div 
                                key={childStructureNode.id} 
                                className='columns is-clickable' 
                                onClick={() => onSelectChild(childStructureNode)}
                            >
                                <div className='column'>
                                    <ObjectName structureNode={childStructureNode} />
                                </div>
                                <div className='column'>
                                    <ObjectPreview structureNode={childStructureNode} />
                                </div>
                            </div>)
                    }</Card.Content>
                </Card>
            </div>}

        </div>
    </div>;
};

GroupView.Path = "/groupview";

export { GroupView };
