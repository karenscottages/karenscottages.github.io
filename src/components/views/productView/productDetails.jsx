import React from 'react';
import { Card } from '/components/atoms/cards';
import { ObjectName, ObjectPreview } from '/agility/structureNode/components';
import { StructureNodeContext } from '/agility/structureNode';
import { StructureAttributeInstances } from './structureAttributeInstances.jsx';

export const ProductDetails = () => {
    const { structureNode } = React.useContext(StructureNodeContext);

    return <Card>
        <Card.Header>
            {structureNode && <ObjectName />}
        </Card.Header>
        <Card.Content>
            {structureNode && <>
                <ObjectPreview />
                <StructureAttributeInstances structureNode={structureNode} />
            </>}
        </Card.Content>
    </Card>;
};
