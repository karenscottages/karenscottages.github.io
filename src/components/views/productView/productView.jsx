import React from 'react';
import { ProductDetails } from './productDetails.jsx';
import { ProductSummary } from './productSummary.jsx';

const ProductView = () => <div className="container is-fluid">
    <div className="section columns is-multiline">

        <div className="column">
            <ProductSummary />
        </div>

        <div className="column">
            <ProductDetails />
        </div>

    </div>
</div>;

ProductView.Path = "/productview";

export { ProductView };
