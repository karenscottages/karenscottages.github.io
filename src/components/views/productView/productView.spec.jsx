import React from 'react';
import renderer from 'react-test-renderer';
import { ProductView } from './productView.jsx';

describe("Views/productView/productView", () => {
    describe("Product View page", () => {
        it('Renders with no props', () => {
            const tree = renderer.
                create(<ProductView/>).
                toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});