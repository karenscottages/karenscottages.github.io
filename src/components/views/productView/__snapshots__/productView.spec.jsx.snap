// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Views/productView/productView Product View page Renders with no props 1`] = `
<div
  className="container is-fluid"
>
  <div
    className="section columns is-multiline"
  >
    <div
      className="column"
    >
      <div
        className="card "
      >
        <header
          className="card-header "
        >
          <div
            className="columns"
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <div
              className="column"
            >
              <p
                className="card-header-title"
              >
                 
                Summary
                 
              </p>
            </div>
          </div>
        </header>
        <div
          className="card-content "
        />
      </div>
    </div>
    <div
      className="column"
    >
      <div
        className="card "
      >
        <header
          className="card-header "
        >
          <div
            className="columns"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </header>
        <div
          className="card-content "
        />
      </div>
    </div>
  </div>
</div>
`;
